nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/LockException.java

```diff
@@ -43,4 +43,34 @@ public LockException(Throwable cause) {
         super(cause);
     }
 
+    /**
+     * @since 11.1
+     */
+    public LockException(int statusCode) {
+        this.statusCode = statusCode;
+    }
+
+    /**
+     * @since 11.1
+     */
+    public LockException(String message, int statusCode) {
+        super(message);
+        this.statusCode = statusCode;
+    }
+
+    /**
+     * @since 11.1
+     */
+    public LockException(String message, Throwable cause, int statusCode) {
+        super(message, cause);
+        this.statusCode = statusCode;
+    }
+
+    /**
+     * @since 11.1
+     */
+    public LockException(Throwable cause, int statusCode) {
+        super(cause);
+        this.statusCode = statusCode;
```

Comment:
Don't set `statusCode` in this class, just delegate to the appropriate `super` that takes a `statusCode`.